core/src/main/java/com/linecorp/armeria/internal/common/util/TemporaryThreadLocals.java

```diff
@@ -15,82 +15,103 @@
  */
 package com.linecorp.armeria.internal.common.util;
 
+import static com.google.common.base.Preconditions.checkState;
+
 import com.google.common.annotations.VisibleForTesting;
+import com.google.errorprone.annotations.MustBeClosed;
 
 import io.netty.util.internal.EmptyArrays;
 
 /**
  * Provides various thread-local variables used by Armeria internally, mostly to avoid allocating
  * short-living objects, such as {@code byte[]} and {@link StringBuilder}. Keep in mind that the variables
- * provided by this class must be used with extreme care, because otherwise it will result in unpredictable
- * behavior.
+ * provided by this class must be used and released with extreme care, because otherwise it will result in
+ * unpredictable behavior.
  *
  * <p>Most common mistake is to call or recurse info a method that uses the same thread-local variable.
  * For example, the following code will produce a garbled string:
  * <pre>{@code
  * > class A {
  * >     @Override
  * >     public String toString() {
- * >         return TemporaryThreadLocals.get().append('"').append(new B()).append('"').toString();
+ * >         try (TemporaryThreadLocals tempThreadLocals = TemporaryThreadLocals.acquire()) {
+ * >             StringBuilder stringBuilder = tempThreadLocals.stringBuilder();
+ * >             return stringBuilder.append('"').append(new B()).append('"').toString();
+ * >         }
  * >     }
  * > }
  * > class B {
  * >     @Override
  * >     public String toString() {
- * >         return TemporaryThreadLocals.get().append("foo").toString();
+ * >         try (TemporaryThreadLocals tempThreadLocals = TemporaryThreadLocals.acquire()) {
+ * >             StringBuilder stringBuilder = tempThreadLocals.stringBuilder();
+ * >             return stringBuilder.append("foo").toString();
+ * >         }
  * >     }
  * > }
- * > // The following assertion fails, because A.toString() returns "foofoo\"".
- * > assert "\"foo\"".equals(new A().toString());
- * }</pre></p>
- *
- * <p>A general rule of thumb is not to call other methods while using the thread-local variables provided by
- * this class, unless you are sure the methods you're calling never uses the same thread-local variables.</p>
+ * }</pre>
+ * If no exception occurs, {@code new A().toString()} returns {@code foofoo"}. However, it does not happen
```

Comment:
```suggestion
 * If no exception occurs, {@code new A().toString()} returns {@code foofoo}. However, it does not happen
```

Reply:
I think we can just completely remove this sentence. 😄

Reply:
Yes, with https://github.com/line/armeria/pull/3505#discussion_r642455211, it does not seem to be needed.